apps/src/containedLevels.js

```diff
@@ -33,6 +33,15 @@ export function getContainedLevelResultInfo() {
 }
 
 /**
+ * We don't report the validated result to the server, since we always treat
+ * contained levels as correct from a progress point of view. We do use it for
+ * displaying feedback though.
+ */
+export function getValidatedResult() {
+  return codeStudioLevels.getContainedLevelResult().result.result;
```

Comment:
The names are rather unfortunate.

The innermost result is just the boolean indicating whether or not the answer was correct. The `result` object around that is all of the result info returned by the mult/free response level, which includes more info like the response itself and testResult. `getContainedLevelResult()` includes the level result and also has the level id and other stuff used for the milestone post.